### PR TITLE
38 summary action bar test fixture

### PR DESCRIPTION
**Pull request by blackbaud-svossen**

### CODE REVIEW

File: e2e/summary-action-bar.e2e-spec.ts [hunk 1]
```diff
@@ -101,7 +101,7 @@ describe('Summary Action Bar', () => {
   it('should match previous tab with a summary action bar screenshot', (done) => {
     SkyHostBrowser.get('visual/tab-summary-action-bar');
     SkyHostBrowser.setWindowBreakpoint('lg');
-    element(by.css('#sky-tab-2-nav-btn')).click();
+    element(by.css('#sky-tab-1-nav-btn')).click();
```
Comment (blackbaud-svossen) [inline e2e/summary-action-bar.e2e-spec.ts:104]: This fixes what seems to be a broken test due to the tab id's switching to using a zero based index. I'll have to follow up with this in stand up if not here.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-38-summary-action-bar-test-fixture](https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-38-summary-action-bar-test-fixture)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-38-summary-action-bar-test-fixture](https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-38-summary-action-bar-test-fixture)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-38-summary-action-bar-test-fixture](https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-38-summary-action-bar-test-fixture)

File: src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+
+    it('should expose primary action button properties', async () => {
+      // modify the defaults
+      testComponent.primaryAction = {
+        buttonText: 'some action',
+        isDisabled: true,
+        click: () => testComponent.primaryActionClicked()
+      };
+      fixture.detectChanges();
+      await fixture.whenStable();
```
Comment (Blackbaud-TrevorBurch) [inline src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts:117]: Is this line needed? This test might be able to not be `async`? I could be wrong - just wanted to ask.

File: src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+      expect(action.buttonText).toEqual(testComponent.primaryAction.buttonText);
+      expect(action.isDisabled).toEqual(testComponent.primaryAction.isDisabled);
+    });
+
+  });
+
+  describe('secondary actions', () => {
+
+    it('should expose secondary action button default properties', async () => {
+      const actionSpy = spyOn(fixture.componentInstance, 'secondaryActionClicked');
```
Comment (Blackbaud-TrevorBurch) [inline src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts:131]: `fixture.componentInstance` should be `testComponent` I believe.

File: src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+
+        // verify the click methods work on the button state
+        await action.click();
+        expect(actionSpy).toHaveBeenCalledWith(i);
+      }
+    });
+
+    it('should expose secondary action button default properties when responsive', async () => {
+      await initiateResponsiveMode(SkyMediaBreakpoints.xs);
+      const actionSpy = spyOn(fixture.componentInstance, 'secondaryActionClicked');
```
Comment (Blackbaud-TrevorBurch) [inline src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts:149]: `fixture.componentInstance` should be `testComponent` I believe.

File: src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+      // verify button state
+      expect(action.buttonText).toEqual(testComponent.primaryAction.buttonText);
+      expect(action.isDisabled).toEqual(testComponent.primaryAction.isDisabled);
+
+      // verify the click methods work on the button state
+      await action.click();
+      expect(actionSpy).toHaveBeenCalled();
+    });
+
+    it('should expose primary action button properties', async () => {
```
Comment (Blackbaud-TrevorBurch) [inline src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts:109]: Could we say `should expose primary action button properties if the properties change`? That seems to be what we are testing. I just want to be clear when looking down the road about what we are testing compared to the firs test.

File: src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+        expect(action.buttonText).toEqual(testComponent.secondaryActions[i].buttonText);
+        expect(action.isDisabled).toEqual(testComponent.secondaryActions[i].isDisabled);
+
+        // verify the click methods work on the button state
+        await action.click();
+        expect(actionSpy).toHaveBeenCalledWith(i);
+      }
+    });
+
+    it('should expose secondary action button properties', async () => {
```
Comment (Blackbaud-TrevorBurch) [inline src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts:165]: Could we say `should expose secondary action button properties if the properties change`? That seems to be what we are testing. I just want to be clear when looking down the road about what we are testing compared to the firs test.

File: src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+        expect(action.isDisabled).toEqual(testComponent.secondaryActions[i].isDisabled);
+      }
+    });
+
+  });
+
+  describe('cancel button', () => {
+
+    it('should expose cancel action button default properties', async () => {
+      const actionSpy = spyOn(fixture.componentInstance, 'cancelActionClicked');
```
Comment (Blackbaud-TrevorBurch) [inline src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts:193]: `fixture.componentInstance` should be `testComponent` I believe.

File: src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+      expect(action.isDisabled).toEqual(testComponent.cancelAction.isDisabled);
+
+      // verify the click methods work on the button state
+      await action.click();
+      expect(actionSpy).toHaveBeenCalled();
+    });
+
+    it('should expose cancel action button default properties when responsive', async () => {
+      await initiateResponsiveMode(SkyMediaBreakpoints.xs);
+      const actionSpy = spyOn(fixture.componentInstance, 'cancelActionClicked');
```
Comment (Blackbaud-TrevorBurch) [inline src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts:207]: `fixture.componentInstance` should be `testComponent` I believe.

File: src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+      // verify button state
+      expect(action.buttonText).toEqual(testComponent.cancelAction.buttonText);
+      expect(action.isDisabled).toEqual(testComponent.cancelAction.isDisabled);
+
+      // verify the click methods work on the button state
+      await action.click();
+      expect(actionSpy).toHaveBeenCalled();
+    });
+
+    it('should expose cancel action button properties', async () => {
```
Comment (Blackbaud-TrevorBurch) [inline src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts:219]: Could we say `should expose cancel action button properties if the properties change?` That seems to be what we are testing. I just want to be clear when looking down the road about what we are testing compared to the firs test.

File: src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts [hunk 10]
```diff
@@ -0,0 +1,284 @@
...
+
+  describe('summary content', () => {
+
+    it('should expose content body on large screen', async () => {
+      // verify the content matches
+      const summaryBody = SkyAppTestUtility.getText(summaryActionBarFixture.summaryBody);
+      expect(summaryBody).toEqual(testComponent.summaryBody);
+    });
+
+    it('should expose content body when responsive', async () => {
```
Comment (Blackbaud-TrevorBurch) [inline src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts:245]: I believe you can get the content body even when it is not visible. Should we have a test for that?

**Review by Blackbaud-TrevorBurch — CHANGES_REQUESTED** [review]
@blackbaud-svossen Just few few comments. Most of them - if not all - are style things.

File: src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts (hunk 2 above)
Comment (blackbaud-svossen) [reply src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts:117]: I'm sure you could get away with not waiting for the fixture to stabilize, but I'd rather avoid introducing any timing issues based on an aversion to using async. This seems more explicit to me; wait for the changes to take place before proceeding.

File: src/app/public/testing/summary-action-bar/summary-action-bar.module.ts [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+} from '@angular/platform-browser/animations';
+
+import {
+  SkySummaryActionBarModule
+} from '@skyux/action-bars';
+
+@NgModule({
+  exports: [
+    SkySummaryActionBarModule,
+
```
[the remark this answers is not in the document]
Comment (blackbaud-svossen) [reply src/app/public/testing/summary-action-bar/summary-action-bar.module.ts:16]: I cringe seeing no new line above comments. The extra line is to draw attention to the comment block above the Noop module.

File: src/app/public/testing/summary-action-bar/summary-action-bar-fixture.ts [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+
+      this.fixture.detectChanges();
+      return this.fixture.whenStable();
+    }
+  }
+
+  /**
+   * Toggles the summary content area open or closed, if in responsive mode.
+   */
+  public async toggleSummaryContent(): Promise<void> {
```
[the remark this answers is not in the document]
Comment (blackbaud-svossen) [reply src/app/public/testing/summary-action-bar/summary-action-bar-fixture.ts:98]: It seems pretty clear to me. If you were to follow this everywhere else you'd need to change the `toggleSecondaryActionDropdown` as well.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-38-summary-action-bar-test-fixture](https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-38-summary-action-bar-test-fixture)

File: src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts (hunk 2 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts:117]: I get what you're saying. I think here though we are just changing some values so there really isn't anything to stabilize as changing those values does not start any async operations. Angular's change detection will happen via the `detectChanges` call and we are just changing bindings and so there isn't anything to stabilize. I also think that with this being in the `spec` that we don't need to worry about what might happen for others - if it works without it then it works. We aren't having to deal with any consumer code that might be wonky in this case like we might be dealing with in the fixture itself.

File: src/app/public/testing/summary-action-bar/summary-action-bar.module.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (Blackbaud-TrevorBurch) [reply src/app/public/testing/summary-action-bar/summary-action-bar.module.ts:16]: Lets chat about this in flex time. I get what you are saying - but we don't have precedence anywhere in our codebase that I know of. I just want to make sure we are being consistent.

**Comment by Blackbaud-TrevorBurch** [comment]
@blackbaud-svossen the build is failing on the new test.

File: src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts (hunk 2 above)
Comment (blackbaud-svossen) [reply src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts]: Sure

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-38-summary-action-bar-test-fixture](https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-38-summary-action-bar-test-fixture)

**Comment by blackbaud-svossen** [comment]
@Blackbaud-TrevorBurch 

I'm also noticing that the PR build is failing due to coverage issues. The coverage issue was present before my PR as well.

I'd like to know if there is a way around this since it has nothing to do with the PR. In my book this should have never been able to have happened with the CI/CD set up this way. 

I'd suggest we log a bug/story to get this coverage back up to avoid further issues with pushing new PRs.

File: src/app/public/testing/summary-action-bar/summary-action-bar.module.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (Blackbaud-TrevorBurch) [reply src/app/public/testing/summary-action-bar/summary-action-bar.module.ts:16]: Discussed. We like this at the team level. :)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-38-summary-action-bar-test-fixture](https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-38-summary-action-bar-test-fixture)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-38-summary-action-bar-test-fixture](https://developer.blackbaud.com/skyux-action-bars?_sp.spa=pr-38-summary-action-bar-test-fixture)

**Review by Blackbaud-TrevorBurch — APPROVED** [review]

**Review by Blackbaud-TrevorBurch — APPROVED** [review]

**Pull request merged**